language/diem-framework/modules/Diem.move

```diff
@@ -651,8 +651,14 @@ module Diem {
         // `PreburnQueue` resource already, in order to be upgraded.
         if (preburn_exists && !preburn_queue_exists) {
             let Preburn { to_burn } = move_from<Preburn<CoinType>>(sender);
-            destroy_zero(to_burn);
             publish_preburn_queue<CoinType>(account);
+            // If the DD has an old preburn balance, this is converted over
+            // into the new preburn queue when it's upgraded.
+            if (to_burn.value > 0)  {
+                add_preburn_to_queue(account, Preburn { to_burn })
+            } else {
+                destroy_zero(to_burn)
+            };
```

Comment:
@moeziniaf Let me know if these changes look good to you.

This now allows a DD to upgrade even if they have a full preburn from before the Diem Framework was updated (I'll cc you in the test case that demonstrates this).

Reply:
great - this should give us the guarantee there are either 0 or 1 preburns in the Queue after the upgrade.

Reply:
(If we wanted to be extra pedantic, we could make this check for the upgrade case too but again that piece of code `add_preburn_to_queue_first_time` would be a one time piece of code)